src/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.ts

```diff
@@ -0,0 +1,203 @@
+		return {
+			...DEFAULT_TEXT_QUERY_BUILDER_OPTIONS,
+			... {
+				extraFileResources: this._instantiationService.invokeFunction(getOutOfWorkspaceEditorResources),
+				maxResults: this.searchConfig.maxResults ?? undefined,
+				isSmartCase: this.searchConfig.smartCase,
+			},
+
+			previewOptions: {
+				matchLines: 1,
+				charsPerLine
+			}
+		};
+	}
+
+	constructor(
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IWorkspaceContextService private readonly _contextService: IWorkspaceContextService,
+		@IEditorService private readonly _editorService: IEditorService,
+		@ILabelService private readonly _labelService: ILabelService,
+		@IViewsService private readonly _viewsService: IViewsService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+	) {
+		super(TEXT_SEARCH_QUICK_ACCESS_PREFIX, { canAcceptInBackground: true });
+
+		this.queryBuilder = this._instantiationService.createInstance(QueryBuilder);
+		this.searchModel = this._instantiationService.createInstance(SearchModel);
+	}
+
+	private get searchConfig(): ISearchConfigurationProperties {
+		return this._configurationService.getValue<ISearchConfigurationProperties>('search');
+	}
+
+	private async doSearch(contentPattern: string): Promise<SearchResult | undefined> {
+
+		if (contentPattern === '') {
+			return undefined;
+		}
+
+		const folderResources: IWorkspaceFolder[] = this._contextService.getWorkspace().folders;
+		const content: IPatternInfo = {
+			pattern: contentPattern,
+		};
+		const charsPerLine = content.isRegExp ? 10000 : 1000; // from https://github.com/microsoft/vscode/blob/e7ad5651ac26fa00a40aa1e4010e81b92f655569/src/vs/workbench/contrib/search/browser/searchView.ts#L1508
+
+		const query: ITextQuery = this.queryBuilder.text(content, folderResources.map(folder => folder.uri), this._getTextQueryBuilderOptions(charsPerLine));
+
+		await this.searchModel.search(query, undefined);
+		return this.searchModel.searchResult;
+	}
+
+	private moveToSearchViewlet(model: SearchModel, currentElem: RenderableMatch) {
+		// this function takes this._searchModel.searchResult and moves it to the search viewlet's search model.
+		// then, this._searchModel will construct a new (empty) SearchResult, and the search viewlet's search result will be disposed.
+		this._viewsService.openView(VIEW_ID, false);
+		const viewlet: SearchView | undefined = this._viewsService.getActiveViewWithId(VIEW_ID) as SearchView;
+		viewlet.importSearchResult(model);
+
+		const viewer: WorkbenchCompressibleObjectTree<RenderableMatch> | undefined = viewlet?.getControl();
+
+		viewer.setFocus([currentElem], getSelectionKeyboardEvent());
+		viewer.setSelection([currentElem], getSelectionKeyboardEvent());
+		viewer.reveal(currentElem);
+	}
+
+	protected async _getPicks(contentPattern: string, disposables: DisposableStore, token: CancellationToken): Promise<(IQuickPickSeparator | IPickerQuickAccessItem)[]> {
```

Comment:
were you doing the FastAndSlow picks later?

Reply:
Yup! There's some extra work with getting sync and async results from the search service, so I decided to merge it in after this.